test/minitest/test_minitest_mock.rb

```diff
@@ -495,4 +495,18 @@ def test_mock_with_yield
     @tc.assert_equal true, rs
   end
 
+  def test_blow_up_on_multiple_stub
+    h = {}
+    e = assert_raises StubError do
+      h.stub(:empty?, false) do
+        h.stub(:empty?, true) do
```

Comment:
There has never been a doubt that this is bad code. I thought the question was "Do we make a specific error message or not?". I didn't mean scenarios where somebody would need this functionality. I meant scenarios where this mistake could happen.

The confusion probably comes from "my test is not good enough". I meant the test for the exceptions. I don't doubt that the nested stub is bad.

And then that is what exceptions are for, otherwise there would not be any "ArgumentError: wrong number of arguments" in ruby.